src/tests_cfg/entity_linked.rs

```diff
@@ -32,3 +33,27 @@ impl Linked for CakeToFillingVendor {
         ]
     }
 }
+
+#[derive(Debug)]
+pub struct CheeseCakeToFillingVendor;
+
+impl Linked for CheeseCakeToFillingVendor {
+    type FromEntity = super::cake::Entity;
+
+    type ToEntity = super::vendor::Entity;
+
+    fn link(&self) -> Vec<RelationDef> {
+        vec![
+            super::cake_filling::Relation::Cake
+                .def()
+                .on_condition(|left, _| {
+                    Expr::tbl(left, super::cake::Column::Name)
+                        .like("%cheese%")
+                        .into_condition()
+                })
+                .rev(),
+            super::cake_filling::Relation::Filling.def(),
+            super::filling::Relation::Vendor.def(),
+        ]
+    }
+}
```

Comment:
I think you nailed that in terms of API ergonomics. And yeah no reason we cannot bake that into RelationDef. The only concern I had was it's kind of complicated to reason about.

And I'd appreciate that it to be documented somewhere what the second parameter is, or at least `left, _right`

Reply:
Actually, does that still work when a `rev` is applied?

Reply:
With `rev` or without `rev`. The behaviour is the same. 
- `left` represent the `from_tbl` inside `RelationDef`
- `right` represent the `to_tbl` inside `RelationDef`

Examples: https://github.com/SeaQL/sea-orm/pull/793/files#diff-be3eeec8b487c64c2f782fc15091165849d2677415f718a648dfcc1221044152R385-R426